common/short-hex-str/src/lib.rs

```diff
@@ -82,15 +82,24 @@ const HEX_CHARS_LOWER: &[u8; 16] = b"0123456789abcdef";
 /// the second character as ASCII bytes.
 #[inline(always)]
 fn byte2hex(byte: u8) -> (u8, u8) {
-    let hi = HEX_CHARS_LOWER[((byte >> 4) & 0x0f) as usize];
+    let hi = HEX_CHARS_LOWER[(byte
+        .checked_shr(4)
+        .expect("Integer Overflow at byte2hex")
```

Comment:
Aah.. I think you misunderstood my last comment here. We indeed want `byte2hex` to return a `Result<(u8, u8), Error>` as you had it before, but in the unit tests, we don't need to return the result when we call this function. We simply just want to call `unwrap()` or `expect()` in the unit test. Does that make sense?

The problem with calling the `expect()` here is that non-test code that relies on this function will panic, which isn't great. So, all you need to do is bring back the old logic here and then for the unit tests that call this function, `unwrap()` there. Does that make sense?

The same goes for `hex_encode` below

Reply:
Once you've done that you can call `/canary` on this PR and see how it goes!

Reply:
The two functions originally return u8 tuple and nothing, and the previous changes serve as a temporary error handling measure to line up with error handling in other lib files; changing their return type to Result type at this point might cause all codes that call these two functions to break. I'll commit the other two files for now